packages/entities/entities-gateway-services/src/components/GatewayServiceForm.vue

```diff
@@ -587,34 +672,248 @@ const changeCheckedGroup = () => {
   form.fields.tls_verify_value = formFieldsOriginal.tls_verify_value
 }
 
-const validateUrl = (): void => {
-  if (form.fields.url && checkedGroup.value === 'url') {
+const changeCheckedGroup = () => {
+  isCollapsed.value = true
+  resetFormFieldErrors()
+  form.errorMessages = []
+  initFieldDefaultValues()
+}
+
+const handleTrySampleApi = (): void => {
+  // Increment demo try counter
+  demoTryTimes.value += 1
+
+  // Determine which demo service to use based on even/odd counter
+  const isWeatherDemo = demoTryTimes.value % 2 === 0
+  const serviceRecord = isWeatherDemo ? { ...demoWeatherService } : { ...demoStockService }
+
+  // Reset form fields before populating
+  initFieldDefaultValues()
+
+  form.fields.name = serviceRecord.name
+
+  switch (checkedGroup.value) {
+    case 'url':
+      // Construct the full URL
+      form.fields.url = `${serviceRecord.protocol}://${serviceRecord.host}${serviceRecord.path}`
+      break
+
+    case 'protocol':
+      // Populate individual protocol-related fields
+      form.fields.host = serviceRecord.host
+      form.fields.path = serviceRecord.path
+      form.fields.protocol = serviceRecord.protocol
+      form.fields.port = serviceRecord.port
+      break
+
+    default:
+      console.warn('Invalid input group selected')
```

Comment:
it shouldn't be possible to get here? If checkedGroup is somehow something else, let's just fill whichever view will be displayed by default (`url`)